src/layout/Summary2/SummaryComponent2/TaskSummary.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState } from 'react';
+
+import { Accordion } from '@digdir/designsystemet-react';
+
+import { FormProvider } from 'src/features/form/FormContext';
+import { useLayoutSets } from 'src/features/form/layoutSets/LayoutSetsProvider';
+import { useLanguage } from 'src/features/language/useLanguage';
+import { ComponentSummary } from 'src/layout/Summary2/SummaryComponent2/ComponentSummary';
+import { PageSummary } from 'src/layout/Summary2/SummaryComponent2/PageSummary';
+import { useTaskStore } from 'src/layout/Summary2/taskIdStore';
+import { useNodes } from 'src/utils/layout/NodesContext';
+
+interface TaskSummaryProps {
+  taskId: string;
+  pageId?: string;
+  componentId?: string;
+  summaryOverrides: any;
+}
+
+function TaskSummaryAccordion({ pageKey, children }: React.PropsWithChildren<{ pageKey: string }>) {
+  const [isOpen, setIsOpen] = useState(true);
+  const { langAsString } = useLanguage();
+  return (
+    <Accordion
+      border
+      color={'neutral'}
+    >
+      <Accordion.Item
+        key={pageKey}
+        open={isOpen}
+      >
+        <Accordion.Header onHeaderClick={() => setIsOpen(!isOpen)}>{langAsString(pageKey)}</Accordion.Header>
+        <Accordion.Content>{children}</Accordion.Content>
+      </Accordion.Item>
+    </Accordion>
+  );
+}
+
+function TaskSummary({ pageId, componentId, summaryOverrides }: TaskSummaryProps) {
+  const nodes = useNodes();
+  if (componentId) {
+    const nodeToRender = nodes.findById(componentId);
+    return (
+      nodeToRender && (
+        <ComponentSummary
+          componentNode={nodeToRender}
+          summaryOverrides={summaryOverrides}
+        />
+      )
+    );
+  }
+
+  let pageKeys = nodes.allPageKeys();
+
+  if (pageId) {
+    pageKeys = pageKeys.filter((key) => key === pageId);
+  }
+
+  return (
+    <div style={{ width: '100%' }}>
+      {pageKeys.map((page) => (
+        <div
+          style={{ marginBottom: '10px' }}
+          key={page}
+        >
+          <TaskSummaryAccordion pageKey={page}>
```

Comment:
Nice! But are we sure we want to group pages in accordions by default? For some apps that might make sense, but those that don't use the NavigationBar component they don't really have an incentive to add translations for the page keys. I'd make this optional and add an accordion config for those who want it.

Reply:
The message from Edvin is that they should always render in accordion, but Ill ask him about this :)

Reply:
@olemartinorg Can apps that don't use the NavigationBar still have multiple pages, with their own respective titles?

The idea for having each page in it's own accordion is to mirror the structure of the form on the summary page, and to separate these pages more clearly with some visual element. I'm thinking that if an app doesn't show the navigation, it will still have the user going through a series of pages to fill out the form, and the accordions on the summary page will reflect this and make it a little easier to see what content they are reviewing.

Reply:
> Can apps that don't use the NavigationBar still have multiple pages[...]

Yes!

> [...] with their own respective titles?

Maybe. When adding a page, it doesn't have a natural title. The title of the _app_ shows up on all pages, but translating the page _key_ to something user-readable is optional (and I suspect it's rarely done).

> The idea for having each page in it's own accordion is to mirror the structure of the form on the summary page

Does that mean that pointing a `Summary2` to the current layout-set (aka "the form you're in now") also splits it up into one-accordion-per-page? I haven't seen any apps doing that using the old `Summary` component, and I would be hesitant to force apps into that pattern (rather have it be opt-in rather than opt-out).

Designsystemets recommendations are to [show one question per page](https://www.designsystemet.no/monstre/obligatoriske-og-valgfrie-felt#1-ett-sp%C3%B8rsm%C3%A5l-per-side), and if that turns into _one answer per accordion_, we're pushing app developers away from our own recommendations, as it would make sense to cram more things into fewer pages.

> I'm thinking that if an app doesn't show the navigation, it will still have the user going through a series of pages to fill out the form, and the accordions on the summary page will reflect this and make it a little easier to see what content they are reviewing.

My opinion is that a 'summary page' is , at the same time, also an 'overview page' to show you everything you've filled out before submitting, and hiding that content behind a clickable (assuming the accordions are closed by default?) just makes it more difficult to get that overview, IMO. It's always easier to _see the content without having to click anything_, and if we require one click per page to get that overview, you'll need just as many clicks as just browsing through the pages in a `NavigationBar`. So what's the point in having "one summary page" if that page consists of many smaller "accordion pages"?

Reply:
Just an case detail: they are open by default

Reply:
I suppose the need for the accordions arose from apps that has tons of questions on their pages, which a lot of real world apps have. Case in point is the 1008 app from Finanstilsynet which has so much data and questions it would be insane to have one question per page. I really like the recommendation to have one question per page though and in that case I agree it would be strange to show pages in accordions.

So first of all the accordions are open by default. In the case of 1008 the accordions would allow you to easily go through the data on each page and open and close accordions as it suits you.

So how about we simpy make accordions opt-in? If you want accordions you would soon find that you need to add titles just like you would when adding a NavigationBar.

Reply:
I think having the accordions open by default will make sure its still easy to get an overview of the summary. 

But these are some valid points and I agree it sounds like a better idea to then have the accordion layout as an opt-in. It would definitely be more valuable for the larger and more content-heavy forms, but I think you're right about forcing an accordion setup mid-form is a bad idea. 

Good input!

Reply:
This has now been changed to be opt-in :)